Framework/DataHandling/src/LoadPSIMuonBin.cpp

```diff
@@ -356,185 +472,304 @@ void LoadPSIMuonBin::assignOutputWorkspaceParticulars(
   outputWorkspace->getAxis(0)->unit() = lblUnit;
 
   // Set Start date and time and end date and time
-  Mantid::Types::Core::DateAndTime start(
-      getFormattedDateTime(m_header.dateStart, m_header.timeStart));
-  Mantid::Types::Core::DateAndTime end(
-      getFormattedDateTime(m_header.dateEnd, m_header.timeEnd));
+  auto startDate = getFormattedDateTime(m_header.dateStart, m_header.timeStart);
+  auto endDate = getFormattedDateTime(m_header.dateEnd, m_header.timeEnd);
+  Mantid::Types::Core::DateAndTime start(startDate);
+  Mantid::Types::Core::DateAndTime end(endDate);
   outputWorkspace->mutableRun().setStartAndEndTime(start, end);
 
-  // Start the log building
-  Mantid::API::Algorithm_sptr logAlg = createChildAlgorithm("AddSampleLog");
-  logAlg->setProperty("Workspace", outputWorkspace);
-
-  // Log the set temperature
-  logAlg->setProperty("LogType", "String");
-  logAlg->setProperty("LogName", "Set Temperature");
-  logAlg->setProperty("LogText", m_header.temp);
-  logAlg->executeAsChildAlg();
+  addToSampleLog("run_end", startDate, outputWorkspace);
+  addToSampleLog("run_start", endDate, outputWorkspace);
+
+  // Assume unit is at the end of the temperature
+  boost::trim_right(m_header.temp);
+  auto tempUnit = std::string(1, m_header.temp.at(m_header.temp.size() - 1));
+  addToSampleLog("sample_temp_unit", tempUnit, outputWorkspace);
```

Comment:
can we have this in the try catch too? There is little point have the units if the number is not present.

Reply:
The temperature is always put into the sample log, it just depends on whether it is as a string or a number, if the conversion to double doesn't work then it will place it as a string always. So it shouldn't be in the try catch in my opinion because I would have to duplicate it.